test/tests/mesh/mesh_only/tests

```diff
@@ -78,4 +78,13 @@
     issues = '#23386'
     requirement = 'The system shall throw a warning if user-provided extra element id names do not exist on mesh.'
   []
+  [mesh_only_with_incompatible_parameter]
+    type = 'RunException'
+    input = 'mesh_only.i'
+    cli_args = '--check-input --mesh-only'
+    expect_err = "Incompatible command line arguments provided. --check-input cannot be called with mesh_only."
+    method = '!dbg'
+    issues = '#29788'
+    requirement = 'The system shall throw an error is the user provides a parameter that would result in a segfault when used with --mesh-only'
```

Comment:
```suggestion
    requirement = 'The system shall throw an error if the user specifies incompatible command line arguments that would cause a segfault.'
```